data_structures/trie/trie.py

```diff
@@ -46,6 +46,57 @@ def find(self, word: str) -> bool:
             curr = curr.nodes[char]
         return curr.is_leaf
 
+    def match(self, prefix: str) -> [str]:
+        """
+        gets the list of all the words that
+        are followed by certain prefix.
+    
+        long-url: https://www.geeksforgeeks.org/prefix-matching-python-using-pytrie-module/
+        :param prefix: the prefix to be matched
+        :return: List of prefix-matched words.
+
+        """
+        pass
+
+
+        def search(t: TrieNode, prefix: str) -> TrieNode:
+
+            '''
+            Different search function that returns the sub trie
+            instead of True or False
+            :param t: self
+            :param prefix: prefix to be searched
+            :return: a new sub-trie
+            
+            '''
+            curr = t
+            for char in prefix:
+                if char not in curr.nodes:
+                    return False
+                    break
+                curr = curr.nodes[char]
+            return curr
+
+        def dfs(root: TrieNode, s: str, prefix: str, lst):
```

Comment:
`dfs()` -> `depth_first_search()` please.  Doctests?  How do we know this works?